src/libinjection_sqli.c

```diff
@@ -1172,16 +1172,21 @@ static size_t parse_number(struct libinjection_sqli_state *sf) {
         }
     }
 
-    if (have_e == 1 && have_exp == 0) {
-        /* very special form of
-         * "1234.e"
-         * "10.10E"
-         * ".E"
-         * this is a WORD not a number!! */
-        st_assign(sf->current, TYPE_BAREWORD, start, pos - start, cs + start);
-    } else {
+    /* very special form of
+     * "1234.e"
+     * "10.10E"
+     * ".E"
+     *
+     * https://gosecure.ai/blog/2021/10/19/a-scientific-notation-bug-in-mysql-left-aws-waf-clients-vulnerable-to-sql-injection/
+     * From this Bolg, we could see 1.e or 1.E is a risk SQLI, The SQL parser
+     * will ignore it during parsing. Like "1.e(1)" => (1), 1 1.e/1 => 1/1 etc.
+     * So, if a payload such as "1' or 1.e(1)" bypass SQLI detection, which is
+     * really risk, we should detect such SQLI injection in case WAF bypass.
```

Comment:
```suggestion
     * In this blog post, we can see that 1.e or 1.E is a risky SQLI. The SQL parser
     * ignores it during parsing. For example, "1.e(1)" => (1), 1 1.e/1 => 1/1, etc.
     * So, if a payload like "1' or 1.e(1)" bypasses SQLI detection, which is
     * really risky, then we should detect such SQLI injection in case of WAF bypass.
```